### PR TITLE
Refactored Client code

### PR DESCRIPTION
**Pull request by Ai4n**

### CODE REVIEW

File: Client/src/controllers/AdminController.java [hunk 1]
```diff
@@ -87,8 +86,8 @@ private void deleteUser() {
     private ArrayList<User> getAllUserList() {
         GetAllUsersListRequest getAllUsersListRequest = new GetAllUsersListRequest();
         socketController.write(getAllUsersListRequest);
-        jsonMessage = socketController.readUtf();
-        GetAllUsersListResponse getAllUsersListResponse = gson.fromJson(jsonMessage, GetAllUsersListResponse.class);
+        SocketExchange request = socketController.readRequest();
```
Comment (dnlbl) [inline Client/src/controllers/AdminController.java]: rename method name to readMessage()

File: Client/src/controllers/AdminController.java [hunk 2]
```diff
@@ -87,8 +86,8 @@ private void deleteUser() {
...
+        GetAllUsersListResponse getAllUsersListResponse = socketController.readMessage(request.json, new GetAllUsersListResponse());
```
Comment (dnlbl) [inline Client/src/controllers/AdminController.java]: rename method name to convertMessage(..)

File: Server/src/main/controller/ServerController.java [hunk 3]
```diff
@@ -32,22 +31,19 @@ public void run() {
 
     public void handleMessages() {
 
+        //SocketExchange -> write
```
Comment (dnlbl) [inline Server/src/main/controller/ServerController.java]: remove

**Review by dnlbl — CHANGES_REQUESTED** [review]
needs work

**Comment by Ai4n** [comment]
!

**Review by dnlbl — APPROVED** [review]

**Pull request merged**